src/jit/codegenlinear.cpp

```diff
@@ -18,6 +18,104 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 #include "emit.h"
 #include "codegen.h"
 
+//------------------------------------------------------------------------
+// genInitializeRegisterState: Clean the spills' descriptors of "regSet" and mark all the register in "regSet" that hold
+// an argument of the function (which is the first block).
+//
+// Assumptions:
+//    On exit the "rsModifiedRegsMask" (in "regSet") holds all the registers' masks hosting an argument on the function
+//    and elements of "rsSpillDesc" (in "regSet") are setted to nullptr.
+//
+// Notes:
+//    This method is intended to be called only from initializeStructuresBeforeBlockCodeGeneration.
+void CodeGen::genInitializeRegisterState()
+{
+    // Initialize the spill tracking logic
+
+    regSet.rsSpillBeg();
+
+    // If any arguments live in registers, mark those regs as such
+
+    unsigned   varNum;
+    LclVarDsc* varDsc;
+
+    for (varNum = 0, varDsc = compiler->lvaTable; varNum < compiler->lvaCount; varNum++, varDsc++)
+    {
+        // Is this variable a parameter assigned to a register?
+        if (!varDsc->lvIsParam || !varDsc->lvRegister)
+        {
+            continue;
+        }
+
+        // Is the argument live on entry to the method?
+        if (!VarSetOps::IsMember(compiler, compiler->fgFirstBB->bbLiveIn, varDsc->lvVarIndex))
+        {
+            continue;
+        }
+
+        // Is this a floating-point argument?
```

Comment:
To answer the question from @sandreenko earlier, I'm not certain why it skips floating-point arguments. The `RegSet` capabilities used to be more extensive (they were used to do the on-the-fly register allocation for tree temp values, and kept track of which registers were locked, spilled, double-spilled (!) etc.) I suspect an overall cleanup would be warranted, but I don't think it's necessary in order to enable this refactoring.